crates/core/c8y_smartrest/src/operations.rs

```diff
@@ -76,8 +76,32 @@ impl Operations {
         self.operations.retain(|x| x.name.ne(&op_name));
     }
 
-    pub fn try_new(dir: impl AsRef<Path>, cloud_name: &str) -> Result<Self, OperationsError> {
-        get_operations(dir.as_ref(), cloud_name)
+    pub fn try_new(dir: impl AsRef<Path>) -> Result<Self, OperationsError> {
+        get_operations(dir.as_ref())
+    }
+
+    pub fn get_child_ops(
+        ops_dir: impl AsRef<Path>,
+    ) -> Result<HashMap<String, Self>, OperationsError> {
+        let mut child_ops: HashMap<String, Operations> = HashMap::new();
+        let child_entries = fs::read_dir(&ops_dir)
+            .map_err(|_| OperationsError::ReadDirError {
+                dir: ops_dir.as_ref().into(),
+            })?
+            .map(|entry| entry.map(|e| e.path()))
+            .collect::<Result<Vec<PathBuf>, _>>()?
+            .into_iter()
```

Comment:
Why do you `collect` then `iter`? If this is to get rid of the errors, [`flatten`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.flatten) with do the job.

Reply:
I want to collect just the dir entries only. I am not sure how the flattening will help here.